config/src/lib.rs

```diff
@@ -251,8 +251,13 @@ pub fn configuration() -> ConfigHandle {
     CONFIG.get()
 }
 
-pub fn reload() {
-    CONFIG.reload();
+pub enum ConfigFileSelection<'a> {
```

Comment:
This is a nice idea, but there are a couple of cases in the implementation below where I think it would misbehave.  My suggestion would be to not do this and use the CONFIG_FILE_OVERRIDE global that I suggest elsewhere in this review.

However, an alternative way to express that idea is for `CONFIG_FILE_OVERRIDE` to be  `Mutex<ConfigFileSelection>` rather than an `Mutex<Option<PathBuf>>`.  For that to work out, you'd need to make the `FromPath` variant own its data: `FromPath(PathBuf)`.

I'm not sure that it adds much value over `Option<PathBuf>`; it adds more lines and a locally defined type that is equivalent to `Option`, so I think the maintenance cost is slightly higher in the long run, and the `Option` version just wins out for me.

Reply:
I used `Mutex<Option<PathBuf>>` as suggested, and with the config file override terminology it makes a lot of sense :+1: